service/src/main/java/uk/gov/hmcts/reform/fpl/service/RespondentService.java

```diff
@@ -1,92 +1,98 @@
-            .data(data)
-            .build();
+    // expands collection in UI. A value (in this case partyId) needs to be set to expand the collection.
+    private Element<Respondent> emptyElementWithPartyId() {
+        return ElementUtils.element(Respondent.builder()
+            .party(RespondentParty.builder().partyId(randomUUID().toString()).build())
+            .build());
     }
 
-    @SuppressWarnings("unchecked")
-    public AboutToStartOrSubmitCallbackResponse addHiddenValues(CaseDetails caseDetails) {
-        Map<String, Object> data = caseDetails.getData();
-
-        if (caseDetails.getData().containsKey("respondents1")) {
-            List<Map<String, Object>> respondentParties = (List<Map<String, Object>>) data.get("respondents1");
-
-            List<RespondentParty> respondentPartyList = respondentParties.stream()
-                .map(entry -> mapper.convertValue(entry.get("value"), Map.class))
-                .map(map -> mapper.convertValue(map.get("party"), RespondentParty.class))
-                .map(respondent -> {
-                    RespondentParty.RespondentPartyBuilder partyBuilder = respondent.toBuilder();
-
-                    if (respondent.getPartyId() == null) {
-                        partyBuilder.partyId(UUID.randomUUID().toString());
-                        partyBuilder.partyType(PartyType.INDIVIDUAL);
-                    }
-
-                    return partyBuilder.build();
-                })
-                .collect(toList());
-
-            List<Map<String, Object>> respondents = respondentPartyList.stream()
-                .map(item -> ImmutableMap.<String, Object>builder()
-                    .put("id", UUID.randomUUID().toString())
-                    .put("value", ImmutableMap.of(
-                        "party", mapper.convertValue(item, Map.class),
-                        "leadRespondentIndicator", "No"))
-                    .build())
-                .collect(toList());
-
-            data.put("respondents1", respondents);
-        }
+    private Element<Respondent> getElementToAdd(List<Element<Respondent>> confidentialChildren,
+                                                Element<Respondent> element) {
+        return confidentialChildren.stream()
+            .filter(confidentialChild -> confidentialChild.getId().equals(element.getId()))
+            .findFirst()
+            .orElse(element);
+    }
+
+    public List<Element<Respondent>> modifyHiddenValues(List<Element<Respondent>> respondents) {
+        return respondents.stream()
+            .map(element -> {
+                Respondent.RespondentBuilder builder = Respondent.builder();
+
+                if (element.getValue().getParty().getPartyId() == null) {
+                    addHiddenValues(element, builder);
+                } else {
+                    builder.party(element.getValue().getParty().toBuilder().build());
+                }
+
+                if (hiddenContactDetails(element)) {
```

Comment:
if (element.getValue().containsConfidentialDetails()) and the hidden method can be removed.

Reply:
the hidden method can be replaced, I still need a check for containsConfidentialDetails here though no?